_posts/2020-04-11-native-mobile-to-flutter.markdown

```diff
@@ -0,0 +1,114 @@
+---
+layout: post
+title: "Looking Back on Our Migration from Native Mobile to Flutter."
+date: 2020-04-11
+categories: [mobile, flutter]
+author: Bobby Manuel 
+---
+
+Over the last six months, the Ginger mobile team has been busy rewriting our native iOS and Android mobile apps into a shiny new cross-platform Flutter App!  The app has been live in production since February and we are excited to share the incredible gains we have achieved.
+
+### The results are in!
+
+- 2 - 3x improvement in development velocity after the switch.
+- 98% shared code across both platforms.
+- Reduction from 140,000 lines of code to 31,000 lines of code.  This is despite launching with a brand new online/offline content system that did not exist in the native apps.
+- The application polish and user experience is on par with our native iOS application and easily superior to our native android application.
+- Most importantly, we experienced no Flutter related bugs or unpredictable behaviors after the launch.
+
+### How long did this take? 
+
+- Project spanned Aug 2019 - Dec 2019 
+- Initial release to Android users in February 2020.
+- Release to all users Android + iOS March 2020.
+
+### What problems were we hoping to solve?
+
+We had two working native mobile applications, but lacked feature parity for Android and iOS users.
+
+Android and iOS development took place at different paces, with varying quality and cost.
+
+Maintaining redundancy across two mobile teams would have required us to maintain four to six developers rather than as few as two to three in a cross platform environment.
+
```

Comment:
In a way, one of the problems we were tackling was collecting years of technical debt accumulated (probably mostly attributable to Android codebase), in part due to a revolving set of engineers maintaining and evolving the app, but also an app that had evolved in different directions that no longer represented our business' core value proposition.

Reply:
how about ... `Our legacy native apps had accumulated years of technical debt, we were ready for a full rewrite.`